### ISSUE TITLE
trailingClosure enhancement for Quick/Nimble

### ISSUE THREAD

**Issue by escakot** [nicklockwood/SwiftFormat#820]
Hello,

Great job on SwiftFormat but I'm encountering an issue for the below. Fixed by using disabling `trailingClosure`.

```Swift
context("if the port is changed successfully") {
    it("returns success status") {
        viewModel.changePort(to: 1002) { result in
            expect({
                switch result {
                case .success:
                    return .succeeded
                default:
                    return .failed(reason: "Change port failed")
                }
            }).to(succeed())
        }
    }
}
```
The `expect({` gets converted to `expect {`

I saw a special exception for redundantSelf as a fix for `@autoclosure`. As this is widely used as well for Quick/Nimble, would there be a possible enhancement for this?

**Comment by escakot** [comment]
Nevermind. I think it was trailingClosures rule.

**Comment by nicklockwood** [comment]
@escakot I'm unclear what actually needs to be fixed here. What is wrong with converting `expect({` to `expect {`?

**Comment by escakot** [comment]
@nicklockwood Sorry for being unclear.

These are the 2 function declarations by Quick/Nimble:
```
/// Make an expectation on a given actual value. The value given is lazily evaluated.
public func expect<T>(_ expression: @autoclosure @escaping () throws -> T?, file: FileString = #file, line: UInt = #line) -> Expectation<T> {
    return Expectation(
        expression: Expression(
            expression: expression,
            location: SourceLocation(file: file, line: line),
            isClosure: true))
}

/// Make an expectation on a given actual value. The closure is lazily invoked.
public func expect<T>(_ file: FileString = #file, line: UInt = #line, expression: @escaping () throws -> T?) -> Expectation<T> {
    return Expectation(
        expression: Expression(
            expression: expression,
            location: SourceLocation(file: file, line: line),
            isClosure: true))
}
```

When it was using `expect({`, it was using the function with @autoclosure. `expect {` is using the second function. 

Since, this is only for testing. Would there be a way to ignore rules specific to folder? (i.e. Folder containing all the Tests)

Thanks again!

**Comment by nicklockwood** [comment]
@escakot I see. So with the parentheses it passes the closure itself as the expression, but without it it evaluates the closure instead?

I guess in that case I can just add a rule to exempt `expect` from automatic trailing closure conversion.

Thanks for explaining!

**Comment by escakot** [comment]
@nicklockwood Yes, exactly!

Though, I feel that others who write their own functions with `@autoclosure` may encounter this problem. If something similar like `redundantSelf` ability to add `--selfrequired` could be added, that would be great! 

Thanks for looking into this!

**Comment by nicklockwood** [comment]
@escakot fixed in 0.47.6
